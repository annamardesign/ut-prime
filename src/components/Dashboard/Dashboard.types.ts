import {WithStyles, withStyles, createStyles} from '@material-ui/core/styles';
import React from 'react';

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    className?: string;
    tabName: string;
    pageText: string;
}

const styles = createStyles({
    h100pr: {
        height: '100%'
    },
    background: {
        display: 'flex',
        justifyContent: 'space-around',
        height: '100%',
        background: '#FFF'
    },
    marginTop: {
        marginTop: 150
    },
    dashboardBg: {
        margin: '0 auto',
        width: 205,
        height: 136
    },
    dashboardImg: {},
    dashboardText: {
        display: 'block',
        margin: '0 0 30px 0',
        textAlign: 'center',
        color: '#cecece'
    }
});

export const Styled = withStyles(styles);
export type StyledType = React.FC<Props & WithStyles<typeof styles>>
