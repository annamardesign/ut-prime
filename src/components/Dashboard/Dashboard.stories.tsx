import React from 'react';
import { withReadme } from 'storybook-readme';

import Wrap from '../test/wrap';

// @ts-ignore: md file and not a module
import README from './README.md';
import Dashboard from './index';

export default {
    title: 'Dashboard',
    component: Dashboard,
    decorators: [withReadme(README)]
};

export const Basic: React.FC<{}> = () => <Wrap>
    <Dashboard tabName='Dashboard'>
        Dashboard body
    </Dashboard>;
</Wrap>;
