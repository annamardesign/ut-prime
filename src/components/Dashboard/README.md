# Button

## How to use

```jsx
import Dashboard from 'ut-front-devextreme/core/Dashboard';

<Dashboard onClick={handlerOnClick}>Hello Dashboard</Dashboard>;
```

## Props

- **className** - class applied to root `Dashboard` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
