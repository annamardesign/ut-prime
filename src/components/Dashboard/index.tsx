import React from 'react';
import clsx from 'clsx';
import { useTheme } from '@material-ui/core/styles';

import { Portal } from '../Theme';

import { Styled, StyledType } from './Dashboard.types';

const Dashboard: StyledType = ({ classes, className, ...props }) => {
    const {ut} = useTheme<Portal>();
    const theme = {...classes, ...ut.classes};

    return <div className={clsx(theme.h100pr, className)}>
        <div className={theme.background}>
            <div className={theme.marginTop}>
                {props.children}
                <div className={theme.dashboardText}>{ut.portalName} Dashboard</div>
                <div className={clsx(theme.dashboardBg, theme.dashboardImg)} />
            </div>
        </div>
    </div>;
};

export default Styled(Dashboard);
