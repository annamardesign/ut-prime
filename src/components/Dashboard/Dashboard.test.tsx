import React from 'react';

import { render } from '../test';
import Dashboard from './index';

describe('<Dashboard />', () => {
    it('render equals snapshot', () => {
        const { getByTestId } = render(<Dashboard tabName='tab' pageText='page' />);
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
