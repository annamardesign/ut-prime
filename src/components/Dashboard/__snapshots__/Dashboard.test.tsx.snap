// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="Dashboard-h100pr-1"
  >
    <div
      class="Dashboard-background-2"
    >
      <div
        class="Dashboard-marginTop-3"
      >
        <div
          class="Dashboard-dashboardText-6"
        >
          Administration
           Dashboard
        </div>
        <div
          class="Dashboard-dashboardBg-4 Dashboard-dashboardImg-5"
        />
      </div>
    </div>
  </div>
</div>
`;
